File: packages/shared/src/types/branch.ts
export interface Branch {
  name: string;
  type: 'local' | 'remote';
  current: boolean;
  commit: string;
  upstream?: string;
  ahead?: number;
  behind?: number;
}

export interface BranchList {
  current: string;
  all: Branch[];
  local: Branch[];
  remote: Branch[];
}

File: packages/backend.net/Domain/Branch.cs
namespace ForkWeb.Backend.Domain;

public record Branch(
    string Name,
    BranchType Type,
    bool Current,
    string Commit,
    string? Upstream = null,
    int? Ahead = null,
    int? Behind = null
);

public enum BranchType
{
    Local,
    Remote
}

public record BranchList(
    string Current,
    List<Branch> All,
    List<Branch> Local,
    List<Branch> Remote
);
